llvm/include/llvm/Transforms/IPO/SampleProfileMatcher.h

```diff
@@ -19,6 +19,59 @@
 
 namespace llvm {
 
+// Callsite location based matching anchor.
+struct Anchor {
+  LineLocation Loc;
+  FunctionId FuncId;
+
+  Anchor(const LineLocation &Loc, const FunctionId &FuncId)
+      : Loc(Loc), FuncId(FuncId) {}
+  Anchor(const LineLocation &Loc, StringRef &FName) : Loc(Loc), FuncId(FName) {}
+  bool operator==(const Anchor &Other) const {
```

Comment:
Why do we not compare `Loc` as well? 

Except for this special operator== that excludes Loc, this is essentially just a `typedef pair<LineLocation, FunctionId> Anchor`?

Reply:
I see, changed to `using Anchor = std::pair<LineLocation, FunctionId>;` and removed this struct.